File: JavaScript/leetcode_118.js
var generate = function(numRows) {
    let result = [];
    
    for (let i = 0; i < numRows; i++) {
        let row = Array(i + 1).fill(1); // Create row with i+1 elements, all 1

        for (let j = 1; j < i; j++) {
            row[j] = result[i - 1][j - 1] + result[i - 1][j]; // Fill middle values
        }

        result.push(row); // Push once per row
    }

    return result;
}

File: Python/leetcode_118.py
'''
Given an integer numRows, return the first numRows of Pascal's triangle.

In Pascal's triangle, each number is the sum of the two numbers directly
above.
'''


from typing import List


def generate(num_rows: int) -> List[List[int]]:
    result: List[List[int]] = []

    for i in range(num_rows):
        row = [1] * (i + 1)  # Initialize row with 1s
        for j in range(1, i):  # Fill the middle elements
            row[j] = result[i - 1][j - 1] + result[i - 1][j]
        result.append(row)

    return result


print(generate(5))
